### PR TITLE
improve coverage configuration

### PR DESCRIPTION
**Pull request by skarzi**
Use dedicated directory - `.tests_reports/` - for all tests reports,
including `coverage` html, xml and json reports.
Update `coverage` configuration to:

+ measure `branch` coverage, refer to documentation:
  https://coverage.readthedocs.io/en/coverage-5.0.4/branch.html
+ show only not covered files and lines
+ sort report's results by coverage value

I found following configuration more developers friendly because it provide more concise and helpful terminal report and all things from points above.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -88,9 +88,25 @@ addopts =
 # Why do we exclude this file from coverage?
 # Because coverage is not calculated correctly for pytest plugins.
 # And we completely test it anyway.
+branch = True
```
Comment (sobolevn) [inline setup.cfg:91]: We already have that here: https://github.com/wemake-services/django-test-migrations/blob/686d03a6e4898db8b6e49177b82deda916976f45/setup.cfg#L82-L83

File: setup.cfg [hunk 2]
```diff
@@ -88,9 +88,25 @@ addopts =
...
+data_file = .tests_reports/.coverage
 omit =
   django_test_migrations/contrib/pytest_plugin.py
 
+[coverage:report]
+skip_covered = True
```
Comment (sobolevn) [inline setup.cfg:97]: The same is configured here: https://github.com/wemake-services/django-test-migrations/blob/686d03a6e4898db8b6e49177b82deda916976f45/setup.cfg#L81

File: setup.cfg [hunk 3]
```diff
@@ -88,9 +88,25 @@ addopts =
...
+show_missing = True
+sort = Cover
+
+[coverage:xml]
```
Comment (sobolevn) [inline setup.cfg:101]: I guess that this can be added like `--cov-report=xml`

File: .travis.yml [hunk 4]
```diff
@@ -32,7 +32,7 @@ script:
 
 after_success:
   - pip install coveralls
-  - coveralls
+  - coveralls --rcfile=setup.cfg
```
Comment (sobolevn) [inline .travis.yml:35]: Can you please explain why is this required? It seemed to me like it was working before quite well.

File: .gitignore [hunk 5]
```diff
@@ -199,3 +199,4 @@ fabric.properties
 ### Custom ###
 ex.py
 *.sqlite3
+.tests_reports/
```
Comment (sobolevn) [inline .gitignore:202]: What is the reason behind this change?

File: setup.cfg (hunk 1 above)
Comment (skarzi) [reply setup.cfg:91]: ahh, sorry I didn't notice usage of this flags.
However I think it's more intuitive to set all `coverage` related settings in `setup.cfg` `coverage` section and use `--cov*`  only if something cannot be configured by using `coverage` configuration (like for instance `--cov` or `--cov-report`).

I know `pytest-cov` has something like below in its docs: 

> Note that this plugin controls some options and setting the option in the config file will have no effect. These include specifying source to be measured (source option) and all data file handling (data_file and parallel options).

But in most cases `--cov*` flags will override `coverage` settings only if they are actually used.

File: .travis.yml (hunk 4 above)
Comment (skarzi) [reply .travis.yml:35]: I prefer to be explicit if I am not using default values and default value for `coverage` rc file is `.coveragerc` (however in reality it just search for this files by checking several ones).
I can remove it if you want

**Comment by sobolevn** [comment]
Hi @skarzi! 

Thanks again for your proposal. But, I have decided not to include this PR into the project.

There are several key points:
1. It is not a technical change
2. Proposed configuration goes out of sync with https://github.com/wemake-services/wemake-python-package And we have literally tens of packages configured the exact same way. It would be hard for me to keep in mind why this one is special.

However, if you believe that this change is really helpful and has better DX, feel free to reopen this discussion at https://github.com/wemake-services/wemake-python-package This way we can change all our projects to look the same way.

Best,
Nikita

**Comment by skarzi** [comment]
Sure, I will double check if it's worth to change it and eventually add PR to https://github.com/wemake-services/wemake-python-package :+1:
